indico/modules/rb_new/client/js/common/user/selectors.js

```diff
@@ -45,7 +47,18 @@ export const getAllUserRoomPermissions = createSelector(
 
 export const getUnbookableRoomIds = createSelector(
     getAllUserRoomPermissions,
-    permissions => Object.entries(permissions).filter(([, perms]) => !perms.book && !perms.prebook).map(([id]) => +id)
+    ({rooms}) => rooms.rooms,
+    ({bookRoom}) => bookRoom.filters.dates.startDate,
+    (permissions, rooms, startDate) => {
```

Comment:
`getUnbookableRoomIds` is not the right place for this. It's meant to return the room IDs the user can *never* book regardless of date/time.

Reply:
`getUnbookableRoomIds` is used in many places, including the calendar, and there no bookRoom date filter is present so this might break.

Reply:
Does this mean it probably affects the code somewhere else when it shouldn't be?

Reply:
Okay, you replied even before I asked :P

Reply:
You could add a new selector that combines data from `getUnbookableRoomIds` and your extra logic and then use that one when getting the data in the various selectors inside the bookRoom module.